### PR TITLE
feat: SSAPI Auth Token (BPS-281)

### PR DESCRIPTION
**Pull request by Caleb-Hurshman**
### Proposed Change

Adds auth tokens as an authentication option in addition to basic auth.

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

**Comment by Caleb-Hurshman** [comment]
Test failures are because I copied over a test file from the other PR to add to it. The failure should be resolved once the unit test PR gets merged.

File: receiver/splunksearchapireceiver/config.go [hunk 1]
```diff
@@ -51,12 +58,27 @@ func (cfg *Config) Validate() error {
...
 	}
-	if cfg.Username == "" {
+
+	if cfg.Username == "" && cfg.AuthToken == "" {
 		return errors.New("missing Splunk username")
 	}
-	if cfg.Password == "" {
+
+	if cfg.Password == "" && cfg.AuthToken == "" {
 		return errors.New("missing Splunk password")
```
Comment (schmikei) [inline receiver/splunksearchapireceiver/config.go:67]:
```suggestion
		return errors.New("missing Splunk password or auth token")
```

File: receiver/splunksearchapireceiver/receiver.go [hunk 2]
```diff
@@ -111,22 +111,32 @@ func (ssapir *splunksearchapireceiver) Shutdown(ctx context.Context) error {
 
 func (ssapir *splunksearchapireceiver) runQueries(ctx context.Context) error {
 	for _, search := range ssapir.config.Searches {
+		// set default event batch size
+		fmt.Println(search.EventBatchSize)
```
Comment (schmikei) [inline receiver/splunksearchapireceiver/receiver.go:115]:
```suggestion
```

File: receiver/splunksearchapireceiver/receiver.go [hunk 3]
```diff
@@ -111,22 +111,32 @@ func (ssapir *splunksearchapireceiver) Shutdown(ctx context.Context) error {
...
+		if search.EventBatchSize == 0 {
+			search.EventBatchSize = 100
+			fmt.Println(search.EventBatchSize)
```
Comment (schmikei) [inline receiver/splunksearchapireceiver/receiver.go:118]:
```suggestion
```

**Review by schmikei — APPROVED** [review]

**Pull request merged**